src/tools/run-make-support/src/lib.rs

```diff
@@ -245,6 +245,23 @@ pub fn uname() -> String {
     output.stdout_utf8()
 }
 
+/// Inside a glob pattern of files (paths), read their contents and count the
+/// number of regex matches with a given expression (re).
+#[track_caller]
+pub fn count_regex_matches_in_file_glob(re: &str, paths: &str) -> usize {
+    let re = regex::Regex::new(re).expect(format!("Regex expression {re} is not valid.").as_str());
+    let paths = glob::glob(paths).expect(format!("Glob expression {paths} is not valid.").as_str());
+    use io::BufRead;
+    paths
+        .filter_map(|entry| entry.ok())
+        .filter(|entry| entry.as_path().is_file())
+        .filter_map(|path| fs::File::open(&path).ok())
+        .map(|file| io::BufReader::new(file))
+        .flat_map(|reader| reader.lines().filter_map(|entry| entry.ok()))
+        .filter(|line| re.is_match(line))
+        .count()
+}
```

Comment:
Discussion: a globbing API makes me very uneasy, I would strongly prefer if this was a walkdir-style API (and specialized helpers built on such an API, see `src/tools/tidy`'s usage of walkdir-style APIs) instead, because it's not immediately clear if this recursive, deals with "hidden" files, handles non-UTF8, among other filesystem / glob-related possible failstates and gotchas.

Reply:
I have tried my hand at a reworked version of this that does not use glob. Tell me what you think!